lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1112,6 +1112,41 @@ LogicalResult BitsPrimOp::canonicalize(BitsPrimOp op,
         rewriter, op, innerBits.getInput(), newHi, newLo);
     return success();
   }
+  // bits(mux(sel, a, b), ...) => mux(sel, bits(a, ...), bits(b, ...))
+  if (auto mux = dyn_cast_or_null<MuxPrimOp>(inputOp)) {
+    auto bitsHigh = rewriter.create<BitsPrimOp>(op->getLoc(), mux.getHigh(),
+                                                op.getHi(), op.getLo());
+    auto bitsLow = rewriter.create<BitsPrimOp>(op->getLoc(), mux.getLow(),
+                                               op.getHi(), op.getLo());
+    replaceOpWithNewOpAndCopyName<MuxPrimOp>(rewriter, op, mux.getSel(),
+                                             bitsHigh, bitsLow);
+    return success();
+  }
+  // bits(cat(a, b), ...) => bits(a, ...), or bits(b, ...), or cat(bits(a, ...),
+  // bits(b, ...))
+  if (auto cat = dyn_cast_or_null<CatPrimOp>(inputOp)) {
+    if (auto rhsT = cat.getRhs().getType().dyn_cast<IntType>()) {
```

Comment:
I think `Cat` enforces `IntType` parameters so this can be a `cast<IntType>`, and you don't need the surrounding `if`.